src/Modules/Groups/components/Groups/GroupsList.tsx

```diff
@@ -1,6 +1,200 @@
 
 export default function GroupsList() {
+  const [groupsList, setGroupsList] = useState<GroupListResponse[]>([]);
+  const [groupId, setGroupId] = useState("0");
+  const [loading, setLoading] = useState<boolean>(false);
+  const [counterLoading, setCounterLoadind] = useState<number>(0);
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const handleCloseModal = () => setShowModal(false);
+  const handleShowModel = (projectId: string) => {
+    setGroupId(projectId);
+    setShowModal(true);
+  };
+
+  // For Add Group Modal
+  const [showAddUpdateModal, setShowAddUpdateModal] = useState<boolean>(false);
+  const handleOpenAddUpdateModal = () => setShowAddUpdateModal(true);
+  const handleCloseAddUpdateModal = () => setShowAddUpdateModal(false);
+
+  // Function to fetch the list of Groups from the API
+  const getAllGroups = async () => {
+    if (counterLoading == 0) {
+      setLoading(true);
+      setCounterLoadind(1);
+    }
+    try {
+      const response = await axios.get(GROUPS_URLs.getAll, {
+        headers: requestHeader(),
+      });
+      setGroupsList(response.data);
+    } catch (error) {
+      const axiosError = error as AxiosError<AxiosErrorResponse>;
+      toast.error(axiosError.response?.data.message);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  // Function to delete a Group
+  const DeleteGroup = async () => {
+    try {
+      await axios.delete(GROUPS_URLs.delete(groupId), {
+        headers: requestHeader(),
+      });
+      handleCloseModal();
+      getAllGroups();
+      toast.success("Group deleted successfully.");
+    } catch (error) {
+      const axiosError = error as AxiosError<AxiosErrorResponse>;
+      toast.error(
+        axiosError.response?.data.message ||
+          "Failed to delete the Group. Please try again."
+      );
+    }
+  };
+
+  useEffect(() => {
+    getAllGroups();
+  }, []);
+
   return (
-    <div>GroupsList</div>
-  )
+    <>
+      {loading ? (
+        <LoadingSpinner/>
+      ) : (
+        <>
+          {/* Add Group Modal */}
+          {showAddUpdateModal && (
+            <GroupsData
+              handleCloseAddUpdateModal={handleCloseAddUpdateModal}
+              getAllGroups={getAllGroups}
+            />
+          )}
+
+          {/* DeleteConfirmationModal component displays a confirmation dialog when deleting an item */}
+          <DeleteConfirmationModal
+            title="Group"
+            showModal={showModal}
+            handleCloseModal={handleCloseModal}
+            handleDeleteModal={DeleteGroup}
+          />
+
+          {/* Container for the "Add Group" button, aligned to the right */}
+          <div className="flex w-full justify-end my-5">
+            <button
+              onClick={handleOpenAddUpdateModal}
+              className={`inline-flex items-center gap-2 rounded-full bg-transparent px-4 py-2 text-black shadow-sm focus:relative ${Styles["border-btn"]}`}
+            >
+              <svg
```

Comment:
I suggest making the svg as a jsx component and use it here to avoid cluttering the component with unwanted information.
You can use this tool for that or just do it manually.
https://www.svgviewer.dev/svg-to-react-jsx